### ISSUE TITLE
Tag autocompleter inconsistent interaction

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#8028]
Consistent and predictable interaction with UI is key for a good user experience. Tools that behave in a similar way should have a similar interaction.

The "tags autocompleter" in Gutenberg misses a couple features used by similar tools in Gutenberg itself and in the Classic Editor:

<img width="624" alt="screen shot 2018-07-18 at 17 21 56" src="https://user-images.githubusercontent.com/1682452/42891807-86de9166-8ab0-11e8-945a-efb4ab347a12.png">

- it should close when pressing Escape
- it should cycle through the suggestions when using arrows navigation

Worth noting:
- the tags autocompleter in the Classic Editor closes on Escape and cycles
- all the autocompleters in Gutenberg do the same, e.g.: mentions, slash inserter
- not to mention cycling through options is pretty standard in Gutenberg, see all the various menus
